src/dict.c

```diff
@@ -1396,9 +1396,9 @@ unsigned long dictScanDefrag(dict *d,
 /* Because we may need to allocate huge memory chunk at once when dict
  * expands, we will check this allocation is allowed or not if the dict
  * type has expandAllowed member function. */
-static int dictTypeExpandAllowed(dict *d) {
-    if (d->type->expandAllowed == NULL) return 1;
-    return d->type->expandAllowed(
+static int dictTypeReHashAllowed(dict *d) {
+    if (d->type->reHashAllowed == NULL) return 1;
+    return d->type->reHashAllowed(
```

Comment:
I think "rehash" is one word, so `dictTypeRehashAllowed` is better.

Note that when we're shrinking, we don't need to compute the hash, so it's not really rehashing. Maybe "resize allowed" is better then?